pkg/runtime/runtime.go

```diff
@@ -794,13 +825,35 @@ func (a *DaprRuntime) getTopicRoutes() (map[string]TopicRoute, error) {
 	}
 
 	var subscriptions []runtime_pubsub.Subscription
+
+	// handle app subscriptions
 	if a.runtimeConfig.ApplicationProtocol == HTTPProtocol {
 		subscriptions = runtime_pubsub.GetSubscriptionsHTTP(a.appChannel, log)
 	} else if a.runtimeConfig.ApplicationProtocol == GRPCProtocol {
 		client := runtimev1pb.NewAppCallbackClient(a.grpc.AppClient)
 		subscriptions = runtime_pubsub.GetSubscriptionsGRPC(client, log)
 	}
 
+	// handle declarative subscriptions
+	ds := a.getDeclarativeSubscriptions()
+	for _, s := range ds {
+		skip := false
+
+		// don't register duplicate subscriptions
+		for _, sub := range subscriptions {
+			if sub.Route == s.Route && sub.PubsubName == s.PubsubName && sub.Topic == s.Topic {
+				log.Warnf("two identical subscriptions found (sources: declarative, app endpoint). topic: %s, route: %s, pubsubname: %s",
+					s.Topic, s.Route, s.PubsubName)
+				skip = true
+				continue
```

Comment:
break, as we have found a match, no need to continue to compare with rest of the subscirptions.

Reply:
accepted :) changed in last commit.